packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -351,6 +356,22 @@ class TextPainter {
     _needsLayout = false;
   }
 
+  void layoutToMaxIntrinsicWidth({ double maxWidth }) {
+    if (!_needsLayout && width == maxIntrinsicWidth && width <= maxWidth)
```

Comment:
Yes, you're right.  Maybe I should take out the early exit?  The API for this class needs some serious love.